File: deepspace/src/deepspace/TestP2.java
package deepspace;
import java.util.ArrayList;

public class TestP2 {
    public static void main(String[] args) {
        /*
        Weapon aux = new Weapon("primera",WeaponType.LASER,30);
        Weapon aux2 = new Weapon("segunda",WeaponType.PLASMA,30);
        Weapon aux3 = new Weapon("tercera",WeaponType.MISSILE,30);
        ShieldBooster s1 = new ShieldBooster("primera", 10, 30);
        ShieldBooster s2 = new ShieldBooster("segunda", 20, 30);
        ShieldBooster s3 = new ShieldBooster("tercera", 30, 30);

        // Probamos la clase Hangar
        System.out.println("Probamos la clase Hangar \n");
        int capacidad=10;
        Hangar hangar = new Hangar(capacidad);



        hangar.addWeapon(aux);
        hangar.addWeapon(aux2);
        hangar.addWeapon(aux3);
        hangar.addShieldBooster(s1);
        hangar.addShieldBooster(s2);
        hangar.addShieldBooster(s3);

        Hangar hangar2 = new Hangar(hangar);

        Weapon borrado=hangar2.removeWeapon(0);
        ShieldBooster borrado2 = hangar2.removeShieldBooster(0);

        System.out.println(borrado.toString());
        System.out.println("\n");
        System.out.println(borrado2.toString());
        System.out.println("\n");

        ArrayList<Weapon> w = hangar2.getWeapons();
        ArrayList<ShieldBooster> s = hangar2.getShieldBoosters();

        for(Weapon arma: w){
            System.out.println(arma.toString());
            System.out.println("\n");
        }

        for(ShieldBooster escudo: s){
            System.out.println(escudo.toString());
            System.out.println("\n");
        }
        System.out.println("Elementos " + hangar.getMaxElements());*/
        System.out.println("Probando clase Damage");
        Damage damage = new Damage(15, 20);
        System.out.println(damage.getNShields());
        System.out.println(damage.getNWeapons());
        System.out.println(damage.getWeapons());
        System.out.println(damage.toString());
        Weapon arma1 = new Weapon("laser", WeaponType.LASER, 8);
        Weapon arma2 = new Weapon("misil", WeaponType.MISSILE, 10);
        Weapon arma3 = new Weapon("plasma", WeaponType.PLASMA, 15);
        ArrayList<Weapon> armas = new ArrayList<>();
        armas.add(arma1);
        armas.add(arma2);
        armas.add(arma3);

        ArrayList<WeaponType> arma = new ArrayList<>();
        arma.add(arma1.getType());
        arma.add(arma2.getType());
        Damage damage2 = new Damage(arma, 10);
        ShieldBooster escudo = new ShieldBooster("name", 2, 2);
        ArrayList<ShieldBooster> escudos = new ArrayList<>();
        escudos.add(escudo);
        System.out.println(damage2.getNShields());
        System.out.println(damage2.getNWeapons());
        System.out.println(damage2.getWeapons());
        Damage damage3 = new Damage(damage2);
        System.out.println(damage3.toString());
        System.out.println(damage.hasNoEffect());
        Damage damage4 = new Damage(damage2.adjust(armas, escudos));
        System.out.println("-----------");
        System.out.println(damage2.toString());
        System.out.println(damage4.toString());
        System.out.println("-----------");
        System.out.println(damage.getNWeapons());
        damage.discardWeapon(arma1);
        System.out.println(damage.getNWeapons());
        System.out.println(damage2.getWeapons());
        damage2.discardWeapon(arma1);
        System.out.println(damage2.getWeapons());
        System.out.println(damage.getNShields());
        damage.discardShieldBooster();
        System.out.println(damage.getNShields());


        //Prueba del adjust
        ArrayList<WeaponType> armas2 = new ArrayList<>();
        armas2.add(arma1.getType());
        armas2.add(arma2.getType());
        ArrayList<WeaponType> armas3 = new ArrayList<>();
        armas3.add(arma1.getType());
        armas3.add(arma2.getType());
        armas3.add(arma3.getType());
        armas3.add(arma1.getType());
        ArrayList<ShieldBooster> escudos2 = new ArrayList<>();
        escudos2.add(escudo);

        //Más armas en damage que en los argumentos.

        Damage damage5 = new Damage(armas2,4);

        Damage damage6 = new Damage(armas3,1);

        Damage damage7 = new Damage(0,0);

        Damage damage8 = new Damage(0,5);

        System.out.println(damage5.adjust(armas,escudos2).toString());
        System.out.println(damage6.adjust(armas,escudos2).toString());
        System.out.println(damage7.adjust(armas,escudos2).toString());
        System.out.println(damage8.adjust(armas,escudos2).toString());







    }
}

File: ruby/test_p2.rb
#!/usr/bin/env ruby

require_relative 'CombatResult'
require_relative 'Dice'
require_relative 'GameCharacter'
require_relative 'Loot'
require_relative 'ShieldBooster'
require_relative 'ShotResult'
require_relative 'SuppliesPackage'
require_relative 'Weapon'
require_relative 'WeaponType'
require_relative 'Damage'

class TestP2
  def main
    puts "Probando clase Damage"
    damage=Deepspace::Damage.newNumericWeapons(15,20)
    puts damage.nShields
    puts damage.nWeapons
    puts damage.weapons
    puts damage.to_s

    arma1=Deepspace::Weapon.new("laser",Deepspace::WeaponType::LASER,8)
    arma2=Deepspace::Weapon.new("misil",Deepspace::WeaponType::MISSILE,10)
    arma3=Deepspace::Weapon.new("plasma",Deepspace::WeaponType::PLASMA,15)
    armas=Array.new(arma1,arma2,arma3)

    arma=Array.new(arma1.type,arma2.type)

    damage2=Deepspace::Damage.newSpecificWeapons(arma,10)
    escudo=Deepspace::ShieldBooster.new("name",2,2)
    escudos=Array.new(escudo)

    puts damage2.nShields
    puts damage2.nWeapons
    puts damage2.weapons

    damage3=Deepspace::Damage.newCopy(damage2)
    puts damage3.to_s
    puts damage.hasNoEffect

    damage4=Deepspace::Damage.newCopy(damage2.adjust(armas,escudos))
    puts "----------"
    puts damage2.to_s
    puts damage4.to_s
    puts "----------"

    puts damage.nWeapons
    damage.discardWeapon(arma1)
    puts damage.nWeapons
    puts damage2.weapons
    damage2.discardWeapon(arma1)
    puts damage2.weapons
    puts damage.nShields
    damage.discardShieldBooster
    puts damage.nShields

    #prueba del adjust
    armas2=Array.new(arma1.type,arma2.type)
    armas3=Array.new(arma1.type,arma2.type, arma3.type,arma1.type)
    escudos2=Array.new(escudo)

    damage5=Deepspace::Damage.newSpecificWeapons(armas2,4)
    damage6=Deepspace::Damage.newSpecificWeapons(armas3,1)
    damage7=Deepspace::Damage.newNumericWeapons(0,0)
    damage8=Deepspace::Damage.newNumericWeapons(0,5)

    puts damage5.adjust(armas,escudos2).to_s
    puts damage6.adjust(armas,escudos2).to_s
    puts damage7.adjust(armas,escudos2).to_s
    puts damage8.adjust(armas,escudos2).to_s


  end
end

prueba = TestP2.new()

puts prueba.main